### ISSUE TITLE
progressbar dependency missing

### ISSUE THREAD

**Issue by yee379** [saltstack/salt#25435]
On RHEL6, using EPEL.

```
$ salt blah state.highstate --summary -p
[WARNING ] Failed to open log file, do you have permission to write to /var/log/salt/master?
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
NameError: global name 'progressbar' is not defined
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 349, in salt_main
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/salt.py", line 179, in run
    self._progress_ret(progress, out)
  File "/usr/lib/python2.6/site-packages/salt/cli/salt.py", line 271, in _progress_ret
    self.progress_bar = salt.output.get_progress(self.config, out, progress)
  File "/usr/lib/python2.6/site-packages/salt/output/__init__.py", line 64, in get_progress
    mod='output')['{0}.progress_iter'.format(out)](progress)
  File "/usr/lib/python2.6/site-packages/salt/output/progress.py", line 29, in progress_iter
    widgets = [progressbar.Percentage(), ' ', progressbar.Bar(), ' ', progressbar.Timer(), ' Returns: [', progressbar.Counter(), '/{0}]'.format(progress['minion_count'])]
NameError: global name 'progressbar' is not defined
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.6/site-packages/salt/scripts.py", line 349, in salt_main
    client.run()
  File "/usr/lib/python2.6/site-packages/salt/cli/salt.py", line 179, in run
    self._progress_ret(progress, out)
  File "/usr/lib/python2.6/site-packages/salt/cli/salt.py", line 271, in _progress_ret
    self.progress_bar = salt.output.get_progress(self.config, out, progress)
  File "/usr/lib/python2.6/site-packages/salt/output/__init__.py", line 64, in get_progress
    mod='output')['{0}.progress_iter'.format(out)](progress)
  File "/usr/lib/python2.6/site-packages/salt/output/progress.py", line 29, in progress_iter
    widgets = [progressbar.Percentage(), ' ', progressbar.Bar(), ' ', progressbar.Timer(), ' Returns: [', progressbar.Counter(), '/{0}]'.format(progress['minion_count'])]
NameError: global name 'progressbar' is not defined
```

```
$ sudo yum install -y python-progressbar
```

```
$ salt blah state.highstate --summary -p
100% |##############################################################################################################################| Elapsed Time: 0:00:00 Returns: [1/1]

-------------------------------------------
Summary
-------------------------------------------
# of Minions Targeted: 1
# of Minions Returned: 1
# of Minions Did Not Return: 0
-------------------------------------------
```

**Comment by jfindlay** [comment]
@yee379, thanks for the report.  Salt should output an error saying you need to install progressbar for it to work, and we should also add to the `-p` documentation that this dependency is needed in order for it to work.

**Comment by cachedout** [comment]
Fixed in #25714
